core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -47,6 +47,19 @@
  * Metadata for a table.
  */
 public class TableMetadata implements Serializable {
+
+  /**
+   * Iceberg always use the latest stable format version when creating a new table.
```

Comment:
This isn't correct after this commit. I think it should be "Reserved table property for table format version. Iceberg will default a new table's format version to the latest stable and recommended version."

Also, I think this should note that incomplete or unstable versions cannot be selected using this property. When we add v3, this should not be able to set it.